tests/test_certificates.py

```diff
@@ -58,6 +59,28 @@ def test_calculated_properties(self):
         )
         self.assertRegex(certificate.serial_number, r"[0-9A-F]{2}(?::[0-9A-F]{2}){15}")
 
+    def test_serial_number_zero_padding(self):
+        with open(TEST_FILES / "test.certificate", "rb") as client_certificate_f:
+            certificate = Certificate.objects.create(
+                label="Test certificate",
+                type=CertificateTypes.cert_only,
+                public_certificate=File(client_certificate_f, name="test.certificate"),
+            )
+
+        # Test with a small serial number that needs zero-padding
+        with patch.object(
+            type(certificate.certificate),
+            "serial_number",
+            new_callable=PropertyMock,
+            return_value=0x0A0B1C0D0E0F,
+        ):
+            serial = certificate.serial_number
+            # Should be "0A:0B:1C:0D:0E:0F", not "A0:B1:C0:D0:E0:F"
+            self.assertEqual(serial, "0A:0B:1C:0D:0E:0F")
+            # Verify each byte is exactly 2 hex digits
+            for byte_pair in serial.split(":"):
+                self.assertEqual(len(byte_pair), 2)
```

Comment:
```suggestion
            for nibble_pair in serial.split(":"):
                self.assertEqual(len(nibble_pair), 2)
```

2 hex digits are 1 byte, 1 hex digit is a nibble.

Or 
```suggestion
```

because we just asserted the exact value:
`self.assertEqual(serial, "0A:0B:1C:0D:0E:0F")`

Reply:
Abundance of caution, but you're right, it's more noise than signal.